src/main/kotlin/dsl/skill/SkillsBuilder.kt

```diff
@@ -0,0 +1,18 @@
+package dsl.skill
+
+class SkillsBuilder {
+    private val soft: ArrayList<Skill> = ArrayList()
+    private val hard: ArrayList<Skill> = ArrayList()
+
+    fun soft(skill: String) {
+        soft.add(Soft(skill))
+    }
+
+    fun hard(skill: String) {
+        hard.add(Hard(skill))
+    }
+
+    fun build(): Skills {
+        return Skills(soft, hard)
+    }
```

Comment:
```suggestion
    private val skills: MutableList<Skill> = mutableListOf()

    fun soft(skill: String) {
        skills.add(Soft(skill))
    }

    fun hard(skill: String) {
        skills.add(Hard(skill))
    }

    fun build(): Skills {
        return Skills(skills)
    }
```

`sealed class`는 추상 클래스에요. 추상 클래스라는 점을 이용하면, `soft skill`와 `hard skill`에 대해서 하나의 list에 넣어줄 수 있어요.

그리고, 코틀린에서는 `ArrayList`를 사용하는 것보다는 가변에는 `mutableList`, 불변에는 `List`를 사용하고 있습니다. 😊